tests/src/test/scala/cats/tests/MiniFloatSuite.scala

```diff
@@ -0,0 +1,393 @@
+package cats.tests
+
+import cats.kernel.laws.discipline.{
+  BoundedSemilatticeTests,
+  CommutativeMonoidTests,
+  HashTests,
+  OrderTests,
+  SerializableTests
+}
+import cats.kernel.{BoundedSemilattice, CommutativeMonoid, Hash, Order}
+import cats.laws.discipline.MiniFloat
+import cats.laws.discipline.MiniFloat._
+import cats.laws.discipline.arbitrary._
+import cats.syntax.eq._
+import cats.tests.MiniFloatSuite.FloatArithmeticOp
+import munit.Location
+import org.scalacheck.Prop._
+import org.scalacheck.{Arbitrary, Gen}
+
+class MiniFloatSuite extends CatsSuite {
```

Comment:
There are a _lot_ of tests in here, which I hope is OK. I wanted to be thorough when checking the corner-case floating point behaviours.